mpa/modules/models/heads/supcon_cls_head.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+
+from mmcls.models.builder import HEADS, build_loss
+from mmcls.models.heads.base_head import BaseHead
+
+
+@HEADS.register_module()
+class SupConClsHead(BaseHead):
+    """
+    Supervised Contrastive Learning head for Classification using SelfSL
+    Args:
+        num_classes (int): The number of classes of dataset used for training
+        in_channels (int): The channels of input data from the backbone
+        aux_mlp (dict): A dictionary with the out_channels and optionally the
+                         hid_channels of the auxiliary MLP head.
+        loss (dict): The SelfSL loss: BarlowTwinsLoss (default)
+        topk (set): evaluation topk score, default is (1, )
+    """
+
+    def __init__(self, num_classes: int, in_channels: int, aux_mlp, loss, aux_loss, topk=(1,), init_cfg=None, **kwargs):
+        if in_channels <= 0:
+            raise ValueError(f"in_channels={in_channels} must be a positive integer")
+        if num_classes <= 0:
+            raise ValueError("at least one class must be exist num_classes.")
+
+        if isinstance(topk, int):
+            topk = (topk,)
+        for _topk in topk:
+            assert _topk > 0, "Top-k should be larger than 0"
```

Comment:
Is it okay if _topk is larger than num_classes?

Reply:
Fixed!